c/datasetjson.c

```diff
@@ -2220,133 +2229,377 @@ int decodePercentByte(char *inString, int inLength, char *outString, int *outStr
   return 0;
 }
 
-void respondWithDatasetMetadata(HttpResponse *response) {
-#ifdef __ZOWE_OS_ZOS
+void getDatasetAttributes(JsonBuffer *buffer, char** organization, int* maxRecordLen, int* totalBlockSize, char** recordLength, bool* isBlocked, bool* isPDSE) {
+  ShortLivedHeap *slh = makeShortLivedHeap(0x10000,0x10);
+  char errorBuffer[2048];
+  Json *json = jsonParseUnterminatedString(slh,
+                                             buffer->data, buffer->len,
+                                             errorBuffer, sizeof(errorBuffer));
+
+  if (json) {
+    if (jsonIsObject(json)){
+      JsonObject *jsonObject = jsonAsObject(json);
+      //Get array of datasets
+      JsonArray *datasetArray = jsonObjectGetArray(jsonObject,"datasets");
+      int i = 0;
+      Json *element = jsonArrayGetItem(datasetArray,i);
+      if(element && jsonIsObject(element)) {
+        JsonObject *jsonDatasetObject = jsonAsObject(element);
+        // Get dsorg object
+        JsonObject *dsOrg = jsonObjectGetObject(jsonDatasetObject,"dsorg");
+        *organization = jsonObjectGetString(dsOrg,"organization");
+        *maxRecordLen = jsonObjectGetNumber(dsOrg,"maxRecordLen");
+        *totalBlockSize = jsonObjectGetNumber(dsOrg,"totalBlockSize");
+        *isPDSE = jsonObjectGetBoolean(dsOrg,"isPDSE");
+        // Get recfm object
+        JsonObject *recfm = jsonObjectGetObject(jsonDatasetObject,"recfm");
+        *recordLength = jsonObjectGetString(recfm,"recordLength");
+        *isBlocked = jsonObjectGetBoolean(recfm,"isBlocked");
+      }
+    }
+  }
+  SLHFree(slh);
+}
+
+int setAttributesForDatasetCopy(HttpResponse *response, JsonBuffer *buffer, char* datasetAttributes) {
+  char *organization = NULL;
+  char *recordLength = NULL;
+  char *dsnt = NULL;
+  int maxRecordLen = 0;
+  int totalBlockSize = 0;
+  bool isBlocked = NULL;
+  bool isPDSE = NULL;
+  char recFormat[3];
+  int dirBlock = 0;
+
+  getDatasetAttributes(buffer, &organization, &maxRecordLen, &totalBlockSize, &recordLength, &isBlocked, &isPDSE);
+
+  if(recordLength == "U") {
+    respondWithError(response, HTTP_STATUS_BAD_REQUEST,"Undefined-length dataset");
+    return ERROR_UNDEFINED_LENGTH_DATASET;
+  }
+
+  if(organization == NULL) {
+    respondWithError(response, HTTP_STATUS_BAD_REQUEST,"Source dataset does not exist");
+    return ERROR_DATASET_NOT_EXIST;
+  }
+
+  if(!strcmp(organization, "sequential")) {
+    organization = "PS";
+    dsnt = "";
+  } else {
+    organization = "PO";
+    dsnt = (isPDSE) ? "PDSE" : "PDS";
+    dirBlock = 10;
+    respondWithError(response, HTTP_STATUS_INTERNAL_SERVER_ERROR, "Do not support copy-paste for PDS(E) Dataset");
```

Comment:
might want to put a TODO here so you remember that it needs to be implemented later.